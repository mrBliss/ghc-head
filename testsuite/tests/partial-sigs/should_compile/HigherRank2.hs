{-# LANGUAGE RankNTypes #-}
module HigherRank2 where

foo :: (forall a. [a] -> [a]) -> (_, _ _)
foo x = (x [True, False], x ['a', 'b'])


test :: (([Bool], [Char]), ([Bool], [Char]))
test = (foo reverse, foo tail)
