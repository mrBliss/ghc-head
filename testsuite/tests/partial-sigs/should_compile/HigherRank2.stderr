TYPE SIGNATURES
  foo :: (forall a. [a] -> [a]) -> ([Bool], [Char])
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
